### PR TITLE
Null check for partition value.

### PR DESCRIPTION
**Pull request by anjagruenheid**
## Summary

[Issue](https://github.com/linkedin/openhouse/issues/#nnn)] Briefly discuss the summary of the changes made in this 
pull request in 2-3 lines.

## Changes

- [ ] Client-facing API Changes
- [ ] Internal API Changes
- [x] Bug Fixes
- [ ] New Features
- [ ] Performance Improvements
- [ ] Code Style
- [ ] Refactoring
- [ ] Documentation
- [ ] Tests

For all the boxes checked, please include additional details of the changes made in this pull request.  

## Testing Done

- [ ] Manually Tested on local docker setup. Please include commands ran, and their output.
- [x] Added new tests for the changes made.
- [ ] Updated existing tests to reflect the changes made.
- [ ] No tests added or updated. Please explain why. If unsure, please feel free to ask for help.
- [ ] Some other form of testing like staging or soak time in production. Please explain.

For all the boxes checked, include a detailed description of the testing done for the changes made in this pull request.

# Additional Information

- [ ] Breaking Changes
- [ ] Deprecations
- [ ] Large PR broken into smaller PRs, and PR plan linked in the description.

For all the boxes checked, include additional details of the changes made in this pull request.

### CODE REVIEW

**Review by sumedhsakdeo — CHANGES_REQUESTED** [review]
Thanks Anja for the bug fix. Can we add a unit test that fails with this bug prior to fix and passes after the fix.

File: libs/datalayout/src/main/java/com/linkedin/openhouse/datalayout/datasource/TablePartitionStats.java [hunk 1]
```diff
@@ -39,7 +39,9 @@ public PartitionStat call(Row row) {
       Row partition = row.getStruct(0);
       if (partition != null) {
         for (int i = 0; i < partition.size(); i++) {
-          values.add(partition.get(i).toString());
+          if (partition.get(i) != null) {
```
Comment (sumedhsakdeo) [inline libs/datalayout/src/main/java/com/linkedin/openhouse/datalayout/datasource/TablePartitionStats.java:42]: What is the behavior we want for such null partition values? Ignore them, or handle them? Can we discuss pros and cons of either approaches.

Comment (anjagruenheid) [reply libs/datalayout/src/main/java/com/linkedin/openhouse/datalayout/datasource/TablePartitionStats.java]: I changed it from ignoring to adding a 'null' value, I think this makes more sense if the point is to collect all values that are in the partition?

**Review by sumedhsakdeo — APPROVED** [review]
Thanks Anja. LGTM. Small nit: please update the description section Testing Done.

File: libs/datalayout/src/main/java/com/linkedin/openhouse/datalayout/datasource/TablePartitionStats.java (hunk 1 above)
Comment (sumedhsakdeo) [reply libs/datalayout/src/main/java/com/linkedin/openhouse/datalayout/datasource/TablePartitionStats.java]: @teamurko what do you think we should do here?

File: libs/datalayout/src/main/java/com/linkedin/openhouse/datalayout/datasource/TablePartitionStats.java [hunk 2]
```diff
@@ -39,7 +39,11 @@ public PartitionStat call(Row row) {
       Row partition = row.getStruct(0);
       if (partition != null) {
         for (int i = 0; i < partition.size(); i++) {
-          values.add(partition.get(i).toString());
+          if (partition.get(i) == null) {
+            values.add("null");
+          } else {
+            values.add(partition.get(i).toString());
+          }
```
Comment (teamurko) [inline libs/datalayout/src/main/java/com/linkedin/openhouse/datalayout/datasource/TablePartitionStats.java:46]:
```suggestion
         values.add(Objects.toString(partition.get(i)));
```

**Review by teamurko — CHANGES_REQUESTED** [review]
lgtm, thank you for adding a unit-test! can we use Objects.toString, it handles null

**Comment by anjagruenheid** [comment]
> lgtm, thank you for adding a unit-test! can we use Objects.toString, it handles null

Integrated, thanks for the review!

**Review by teamurko — APPROVED** [review]

**Review by teamurko — APPROVED** [review]

**Pull request merged**